fury/ui.py

```diff
@@ -3777,7 +3777,25 @@ def _setup(self):
                               self.down_button_callback)
 
     def resize(self, size):
-        pass
+        """
+            Resizes ListBox2D i.e resizes the UI components which renders ListBox2D, such as slots, scrollbar etc.
+
+        Parameters
+        ----------
+        size : (int, int)
+            New width and height in pixels.
+        """
+        
+        # Resizing the panel_size will resize the slots accordingly. 
+        self.panel.resize(size)
+
+        # Resizing the Scroll Bar.
+        self.scroll_bar.resize(size)
+
+        # Adjusting the Font Size & Line Spacing for "Testing" purpose.
+        self.font_size = self.font_size//2
+        self.line_spacing = self.line_spacing//2
```

Comment:
@SanjayMarreddi Simply calling the resize method wont help. Individual components need to be resized according to their proportions. Try to understand what sub components ListBox2D has and then resize them accordingly. Understand how `TabPanel2D`'s resize method is implemented.

Reply:
@Nibba2018 Actually, I implemented in this manner after understanding `TabPanel2D`'s resize method only.

The below code shows Implementation of `resize` in `TabPanel2D`
```
    def resize(self, size):
        """ Resizes Tab panel.

        Parameters
        ----------
        size : (int, int)
            New width and height in pixels.
        """
        self._text_size = (int(0.7 * size[0]), size[1])
        self._button_size = (int(0.3 * size[0]), size[1])
        self.panel.resize(size)
        self.text_block.resize(self._text_size)
```

So I thought, 
- `resize` option works for Panel Resizing.
- Similarly for Resizing the Scrollbar since it is a `Rectangle2D` UI & resizing method is already implemented for it. 
- Coming to `font_size` & `line_spacing` we need to resize according to their proportions.

Since you suggested, I will try to understand the sub components again & try resizing in other way.

Reply:
Hi @SanjayMarreddi,

Thank you for doing this. Your logic makes sense. It will be good to have some tests to validate your thought and your code. Also, it will help you to catch any errors. It is also good to have a screenshot, before the function call and after the call, so that we can see the effect.

Reply:
@skoudoro

Actually, For testing that function, I used this Tutorial on [ListBox](https://fury.gl/latest/auto_tutorials/02_ui/viz_ui_listbox.html#sphx-glr-auto-tutorials-02-ui-viz-ui-listbox-py) and added this line 
```
listbox.resize((100,100))
```

after the line 
```
values = ["Welcome", "Bye", "Fury"]
listbox = ui.ListBox2D(
    values=values, position=(10, 300), size=(200, 200), multiselection=False
)
```
**I ran the Tutorial, *with and without* the Line for Resizing `listbox.resize((100,100))` being added. 
I compared the UI but Unfortunately there is no RESIZING being happening there. Is there a mistake in the way I am testing ?
If not, Could you please suggest me few ways to find out the bug ?**

Thanks !

Reply:
@Nibba2018 Could you please review it. I have made the necessary changes & it is working right now.

Reply:
@Nibba2018 Could you please review it. I have made the necessary modifications and It is currently working fine.
Please find the attached screenshots.

I have tested this on the Tutorial of [ListBox](https://fury.gl/latest/auto_tutorials/02_ui/viz_ui_listbox.html#sphx-glr-auto-tutorials-02-ui-viz-ui-listbox-py). 

**Before the Function Call** 
![Screenshot (704)](https://user-images.githubusercontent.com/57671048/104909196-8590b680-59ad-11eb-82cf-01b267e981af.png)

The following is a result of
```
listbox = listbox.resize((400,400))
```
**After the Function Call**
![Screenshot (703)](https://user-images.githubusercontent.com/57671048/104909282-a2c58500-59ad-11eb-9458-46e32b4a8965.png)
 
Thanks !!

Reply:
@SanjayMarreddi Before reviewing your PR, I would like you to write tests for your changes. Thanks!

Reply:
@Nibba2018 Sure, I will do that, Thanks!